tests/robot-tests/tests/libs/slack.py

```diff
@@ -9,15 +9,15 @@
 PATH = f'{os.getcwd()}{os.sep}test-results'
 
 
-def _generate_slack_attachments():
+def _generate_slack_attachments(env: str):
     failed_tests = dict()
     passed_tests = dict()
 
-    report = open(f'{PATH}{os.sep}output.xml', 'rb')
-    contents = report.read()
-
-    soup = BeautifulSoup(contents, 'xml')
-    test = soup.find('total').find('stat')
+    with open(f'{PATH}{os.sep}output.xml', 'rb') as report:
+        contents = report.read()
+        soup = BeautifulSoup(contents, 'xml')
+        test = soup.find('total').find('stat')
```

Comment:
These two lines can be moved outside the `with` scope. We only need the `report` file handler for the `.read()`.